File: Frontend/src/types/cliente.ts
export interface Cliente {
  idCliente: number;
  nombre: string;
  apellido: string;
  correo: string;
  telefono: string;
  estado?: "Activo" | "Inactivo";
  fechaRegistro?: string;
  gastoTotal?: string;
  usuario?: string;
  rol?: string;
}

export interface ClienteBackend {
  idCli?: number;
  idCliente?: number;
  nomCli?: string;
  nombre?: string;
  apeCli?: string;
  apellido?: string;
  correoCli?: string;
  correo?: string;
  telCli?: string;
  telefono?: string;
  estado?: string;
  fechaReg?: string;
  fechaRegistro?: string;
  usuario?: string;
  rol?: string;
  credencial?: {
    usuario: string;
    rol: string;
  };
}

File: Backend/src/main/java/com/magicGroup/backend/model/usuarios/Cliente.java

package com.magicGroup.backend.model.usuarios;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "Cliente")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idCli;

    @Column(nullable = false, unique = true, length = 20)
    private String codCli;

    @Column(nullable = false, length = 45)
    private String nomCli;

    @Column(nullable = false, length = 45)
    private String apeCli;

    @Column(nullable = false, unique = true, length = 45)
    private String correoCli;

    @Column(length = 20)
    private String telCli;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Estado estado = Estado.activo;

    @Column(nullable = false)
    private LocalDate fechaReg;

    @OneToOne
    @JoinColumn(name = "id_cred")
    private Credencial credencial;

    public enum Estado {
        activo, inactivo
    }
}
